src/users/infra/routes/userRouter.ts

```diff
@@ -0,0 +1,15 @@
+import { Request, Response, Router } from 'express';
+import { createUserController } from '../../useCases/createUser';
+import { validateJwt } from '../../authService/middleware/auth0.middleware';
+
+const userRouter = Router();
+
+userRouter.post('/users', async (req: Request, res: Response) => createUserController.execute(req, res));
```

Comment:
`import { Request, Response, NextFunction } from 'express';

function errorHandler(
  err: Error,
  req: Request,
  res: Response,
  next: NextFunction
) {
  console.error(err.stack);

  if (err instanceof YourCustomError) {
    res.status(400).json({ error: err.message });
  } else {
    res.status(500).json({ error: 'Something went wrong!' });
  }
}

export default errorHandler;`

`userRouter.post('/users', (req: Request, res: Response, next: NextFunction) => {
  createUserController
    .execute(req, res)
    .catch(next); // Catch and propagate errors to the error-handling middleware
});`

Reply:
@thetnaingtta you will create Logger module right? :D